sensor-gateway/sensor-communication/KvaserCommunicationErrorFactory.h

```diff
@@ -0,0 +1,219 @@
+            errorMessage = canERR_INTERNAL_MESSAGE;
+        } else if (errorCode == canERR_NO_ACCESS) {
+            category = CONNECTION_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_NO_ACCESS_MESSAGE;
+        } else if (errorCode == canERR_NOT_IMPLEMENTED) {
+            category = EXTERNAL_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_NOT_IMPLEMENTED_MESSAGE;
+        } else if (errorCode == canERR_DEVICE_FILE) {
+            category = EXTERNAL_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_DEVICE_FILE_MESSAGE;
+        } else if (errorCode == canERR_HOST_FILE) {
+            category = EXTERNAL_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_HOST_FILE_MESSAGE;
+        } else if (errorCode == canERR_DISK) {
+            category = UNHANDLED_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_DISK_MESSAGE;
+        } else if (errorCode == canERR_CRC) {
+            category = COMMUNICATION_ERROR;
+            severity = CRITICAL;
+            errorMessage = canERR_CRC_MESSAGE;
+        } else if (errorCode == canERR_CONFIG) {
+            category = UNHANDLED_ERROR;
+            severity = CRITICAL;
+            errorMessage = canERR_CONFIG_MESSAGE;
+        } else if (errorCode == canERR_MEMO_FAIL) {
+            category = UNHANDLED_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_MEMO_FAIL_MESSAGE;
+        } else if (errorCode == canERR_SCRIPT_FAIL) {
+            category = EXTERNAL_ERROR;
+            severity = EMERGENCY;
+            errorMessage = canERR_SCRIPT_FAIL_MESSAGE;
+        } else if (errorCode == canERR_SCRIPT_WRONG_VERSION) {
+            category = EXTERNAL_ERROR;
+            severity = WARNING;
+            errorMessage = canERR_SCRIPT_WRONG_VERSION_MESSAGE;
+        } else if (errorCode == canERR_SCRIPT_TXE_CONTAINER_VERSION) {
+            category = EXTERNAL_ERROR;
+            severity = EMERGENCY;
+            errorMessage =
+                    canERR_SCRIPT_TXE_CONTAINER_VERSION_MESSAGE;
+        } else if (errorCode == canERR_SCRIPT_TXE_CONTAINER_FORMAT) {
+            category = EXTERNAL_ERROR;
+            severity = ERROR;
+            errorMessage =
+                    canERR_SCRIPT_TXE_CONTAINER_FORMAT_MESSAGE;
+        } else if (errorCode == canERR_BUFFER_TOO_SMALL) {
+            category = UNHANDLED_ERROR;
+            severity = ERROR;
+            errorMessage = canERR_BUFFER_TOO_SMALL_MESSAGE;
+        } else if (errorCode == canERR__RESERVED) {
+            category = UNRECOGNIZED_ERROR;
+            severity = WARNING;
+            errorMessage = canERR__RESERVED_MESSAGE;
+        } else {
+            category = UNRECOGNIZED_ERROR;
+            severity = CRITICAL;
+            errorMessage = "Error unknown for canlib";
+        }
+        SensorAccessLinkError error(origin, category, severity, errorCode, errorMessage);
+        throw error;
```

Comment:
Clion dit "Throw expression should throw anonymous temporary values instead"...je ne comprend pas trop pouquoi, mais si ca marche on peut laisser tomber ce warning.

Reply:
:-1: